cdap-app-templates/cdap-etl/cdap-data-streams-base/src/test/java/io/cdap/cdap/datastreams/DataStreamsTest.java

```diff
@@ -256,6 +255,46 @@ private RunId testLineageWithMacro(ApplicationManager appManager,
     return RunIds.fromString(sparkManager.getHistory().iterator().next().getPid());
   }
 
+  @Test
+  public void testSourceMacrosWithAllowFlag() throws Exception {
+    Schema schema = Schema.recordOf("test", Schema.Field.of("id", Schema.of(Schema.Type.STRING)));
+    List<StructuredRecord> input = new ArrayList<>();
+    input.add(StructuredRecord.builder(schema).set("id", "123").build());
+
+    String outputName = UUID.randomUUID().toString();
+    DataStreamsConfig etlConfig = DataStreamsConfig.builder()
+        .addStage(new ETLStage("source",
+            MockSource.getPlugin(schema, input, 1L, "${logicalStartTime()}")))
+        .addStage(new ETLStage("sink", MockSink.getPlugin(outputName)))
+        .setBatchInterval("1s")
+        .addConnection("source", "sink")
+        .build();
+
+    ApplicationId appId = NamespaceId.DEFAULT.app("sourceMacroTest");
+    AppRequest<DataStreamsConfig> appRequest = new AppRequest<>(APP_ARTIFACT, etlConfig);
+    ApplicationManager appManager = deployApplication(appId, appRequest);
+
+    Map<String, String> runtimeArgs = new HashMap<>();
+    runtimeArgs.put(io.cdap.cdap.etl.common.Constants.CDAP_STREAMING_ALLOW_SOURCE_MACROS, "true");
```

Comment:
Do we have a already have a test for when the arg is not present?

Reply:
I thought we did but can't find one. Will add one